sympy/matrices/expressions/matexpr.py

```diff
@@ -569,6 +569,84 @@ def recurse_expr(expr, index_ranges={}):
         else:
             return remove_matelement(retexpr, first_index, last_index)
 
+    def as_indexed(self):
```

Comment:
SymPy's standard is `_eval_as_Indexed` (target class named concatenated to `_eval_as_`)

Reply:
Would it be the same standard if I wanted to make it externally accessible?

Reply:
You call it with `.rewrite(Index)` ==> `rewrite` tries to look at methods like `_eval_as_Index`.

Reply:
I think that should be `_eval_rewrite_as_Index` (and `_eval_rewrite_as_Indexed` for `.rewrite(Indexed)`).

Reply:
Yes, right.

Reply:
Yes, rewrite(Indexed) is better. Not only is this a standard SymPy thing, but it will work even if the top-level expression isn't an Indexed, because every Basic object has the `rewrite` method and it automatically recurses into subexpressions.